lib/core/include/irods/authentication_plugin_framework.hpp

```diff
@@ -1,123 +1,160 @@
 #ifndef IRODS_AUTHENTICATION_PLUGIN_FRAMEWORK_HPP
 #define IRODS_AUTHENTICATION_PLUGIN_FRAMEWORK_HPP
 
-#include "irods/authCheck.h"
-#include "irods/authPluginRequest.h"
-#include "irods/authRequest.h"
-#include "irods/authResponse.h"
 #include "irods/authenticate.h"
+#include "irods/getRodsEnv.h"
 #include "irods/irods_auth_constants.hpp"
-#include "irods/irods_auth_factory.hpp"
-#include "irods/irods_auth_manager.hpp"
-#include "irods/irods_auth_plugin.hpp"
-#include "irods/irods_kvp_string_parser.hpp"
-#include "irods/irods_native_auth_object.hpp"
-#include "irods/irods_stacktrace.hpp"
-#include "irods/miscServerFunct.hpp"
-#include "irods/msParam.h"
+#include "irods/irods_exception.hpp"
+#include "irods/irods_load_plugin.hpp"
+#include "irods/irods_plugin_base.hpp"
 #include "irods/rcConnect.h"
-#include "irods/rodsDef.h"
-
-#ifdef RODS_SERVER
-#include "irods/rsAuthCheck.hpp"
-#include "irods/rsAuthRequest.hpp"
-#endif // RODS_SERVER
+#include "irods/rodsErrorTable.h"
 
 #include <boost/any.hpp>
 #include <fmt/format.h>
 #include <nlohmann/json.hpp>
 
-#include <openssl/md5.h>
-
-#include <termios.h>
-#include <unistd.h>
-#include <iostream>
-#include <sstream>
+#include <functional>
 #include <string>
 
+/// \file
+
 using json = nlohmann::json;
 
-namespace irods::experimental
+namespace irods::experimental::auth
 {
-    namespace auth
-    {
-        static const std::string flow_complete{"authentication_flow_complete"};
-
-        static const std::string next_operation{"next_operation"};
-    }
+    static const std::string flow_complete{"authentication_flow_complete"};
+    static const std::string next_operation{"next_operation"};
```

Comment:
Should these be `char*`s instead?

Reply:
What is the reasoning?

Are we talking about avoiding memory allocation?

Reply:
Yep

Reply:
`const char* const` is good. Using `const std::string_view` works as well. It's up to you.